Letca_Anna/week_02/Domain/Domain/CreateClientOp/CreateClientOp.cs

```diff
@@ -15,12 +15,16 @@ public class CreateClientOp : OpInterpreter<CreateClientCmd, ICreateClientResult
         public override Task<ICreateClientResult> Work(CreateClientCmd Op, Unit state)
         {
 
-            var (valid, validationMessage) = Op.Validate();
+            var (valid, validationResults) = Op.Validate();
+            string validationMessage = "";
+            validationResults.ForEach(x => validationMessage += x.ErrorMessage);
 
             if (!valid)
                 return Task.FromResult((ICreateClientResult)new ClientNotCreated(validationMessage));
 
-            return Task.FromResult((ICreateClientResult)new ClientCreated(new Client(Op.Uid)));
+            Client newClient = new Client(Op.Uid, Op.Name);
+            Storage.ClientCollection[Op.Uid] = newClient;
```

Comment:
Is there something that guarantees you that this won't throw an exception because of a bad indexing? 
The main point of an operation is to make it deterministic. Regardless of the state or input of my app, when I feed something into an operation, I should get one of the defined results. Anything that ends with an Exception is not an option.